diesel/src/test_helpers.rs

```diff
@@ -60,7 +60,13 @@ pub fn pg_connection() -> PgConnection {
 
 #[cfg(feature = "postgres")]
 pub fn pg_connection_no_transaction() -> PgConnection {
-    PgConnection::establish(&pg_database_url()).unwrap()
+    use crate::connection::statement_cache::strategy::{
```

Comment:
I think I would prefer to keep the tests for this functionality local to the `strategy.rs` file. That means:

* Don't setup a custom strategy for the global tests
* Have a few tests to check all cases (cache vs not cache, cachable statement vs not cacheable statement) in that module. 

That should make the changeset a bit smaller and the tests more focused on the actual desired behavior.